opamp/config_manager.go

```diff
@@ -38,5 +93,39 @@ func (a *agentConfigManager) CreateEffectiveConfigMsg() (*protobufs.EffectiveCon
 // If there is an error, it returns an error.
 // The caller is responsible for sending the updated configuration to the Opamp server.
 func (a *agentConfigManager) Apply(remoteConfig *protobufs.AgentRemoteConfig) (bool, error) {
-	return false, nil
+	remoteConfigMap := remoteConfig.GetConfig().GetConfigMap()
+
+	if remoteConfigMap == nil {
+		return false, nil
+	}
+
+	remoteCollectorConfig, ok := remoteConfigMap[collectorConfigKey]
+
+	if !ok {
+		return false, nil
+	}
+
+	return a.applyRemoteConfig(a.agentConfig, remoteCollectorConfig.GetBody())
+}
+
+// applyRemoteConfig applies the remote config to the agent.
+func (a *agentConfigManager) applyRemoteConfig(currentConfig *remoteControlledConfig, newContents []byte) (changed bool, err error) {
```

Comment:
Is the `currentConfig` as input param needed? Won't it be available as `a.agentConfig`?

Reply:
You are correct. This should have a simple function instead of a receiver method on the struct.